index.js

```diff
@@ -1,61 +1,9 @@
 'use strict';
```

Comment:
I would separate the tool part, aka the next plugin, from the API handler. This is to avoid the client-side code to include the plugin. I had a similar situation in https://github.com/moxystudio/next-intl, in which:

- The runtime library, aka the API handler, is the main exports.
- The Next.js plugin is imported separately, with `@moxy/next-intl/plugin`, meaning I've placed a `plugin.js` root file that maps to the compiled plugin, which must be a Node.js file (uses CommonJS).

Moreover, I would write everything using `ES Modules` and add Babel, just like `next-intl`. While adding Babel seems unnecessary as you are not using any non-official language features, it's required to transform ES Modules to CommonJS. Finally, I would put everything inside `src` as the library is now being compiled.